### PR TITLE
Initial implementation of NSNotification support

### PR DESCRIPTION
**Pull request by kovpas**

### CODE REVIEW

File: Foundation/NSNotification.swift [hunk 1]
```diff
@@ -45,40 +40,125 @@ public class NSNotification : NSObject, NSCopying, NSCoding {
 
 extension NSNotification {
     public convenience init(name aName: String, object anObject: AnyObject?) {
-        NSUnimplemented()
+        self.init(name: aName, object: anObject, userInfo: nil)
     }
 }
 
+private class NSNotificationReceiver : NSObject {
+    private var object: NSObject = NSObject()
```
Comment (phausler) [inline Foundation/NSNotification.swift]: the object should not be a strong reference by the notification center

Comment (kovpas) [inline Foundation/NSNotification.swift]: Agree. Fixed, thanks for noticing this!

File: Foundation/NSNotification.swift [hunk 2]
```diff
@@ -8,78 +8,129 @@
...
+    public init(name aName: String, object anObject: AnyObject?, userInfo aUserInfo: [NSObject : AnyObject]?) {
+        name = aName
+        object = anObject
+        userInfo = aUserInfo
     }
     
     public required init?(coder aDecoder: NSCoder) {
-        NSUnimplemented()
+        /* do not invoke; not a valid initializer for this class */
+        fatalError()
```
Comment (parkera) [inline Foundation/NSNotification.swift]: Why remove the NSUnimplemented here?

File: Foundation/NSNotification.swift [hunk 3]
```diff
@@ -8,78 +8,129 @@
...
     }
     
     public func encodeWithCoder(aCoder: NSCoder) {
-        NSUnimplemented()
+
```
Comment (parkera) [inline Foundation/NSNotification.swift]: and here

**Comment by parkera** [comment]
This looks like a pretty good starting point for further work on this.

**Comment by parkera** [comment]
@phausler should review this further, but in the meantime @kovpas can this be rebased into a single commit?

**Comment by kovpas** [comment]
@parkera I'm struggling a bit with rebasing, as I've already merged recent updates from apple's repository into my private repo. Should be easier to open a new pull request I guess... Would it be ok?

**Comment by phausler** [comment]
yea re-opening is perfectly fine; it is actually nicer to have a squashed commit if possible since it makes the history less complicated.

**Comment by kovpas** [comment]
@phausler Done: https://github.com/apple/swift-corelibs-foundation/pull/98
Closing this one.

Thanks!

**Pull request closed without merging**